scenarios/state_transitions_steady.go

```diff
@@ -0,0 +1,123 @@
+			"and the state-transitions-per-second scenario option to be set. Any iteration configuration is ignored. For " +
+			"example, can be run with: run-scenario-with-worker --scenario state_transitions_steady --language go " +
+			"--embedded-server --duration 5m --option state-transitions-per-second=3",
+		Executor: loadgen.ExecutorFunc(func(ctx context.Context, runOptions loadgen.RunOptions) error {
+			return (&stateTransitionsSteady{runOptions}).run(ctx)
+		}),
+	})
+}
+
+type stateTransitionsSteady struct{ loadgen.RunOptions }
+
+func (s *stateTransitionsSteady) run(ctx context.Context) error {
+	// The goal here is to meet a certain number of state transitions per second.
+	// For us this means a certain number of workflows per second. So we must
+	// first execute a basic workflow (i.e. with a simple activity) and get the
+	// number of state transitions it took
+
+	// TODO(cretz): Note, this is a known naive approach because if workflows are
+	// backed up and/or slow down this won't match.
+
+	if s.Configuration.Duration == 0 {
+		return fmt.Errorf("duration required for this scenario")
+	}
+	stateTransitionsPerSecond := s.ScenarioOptionInt("state-transitions-per-second", 0)
+	if stateTransitionsPerSecond == 0 {
+		return fmt.Errorf("state-transitions-per-second scenario option required")
+	}
+	durationPerStateTransition := time.Second / time.Duration(stateTransitionsPerSecond)
+	s.Logger.Infof(
+		"State transitions per second is %v, which means %v between every state transition",
+		stateTransitionsPerSecond,
+		durationPerStateTransition,
+	)
+
+	// Execute initial workflow and get the transition count
+	workflowParams := kitchensink.WorkflowParams{
+		ActionSet: kitchensink.ActionSet{Actions: []*kitchensink.Action{kitchensink.NopActionExecuteActivity}},
+	}
+	workflowRun, err := s.Client.ExecuteWorkflow(
+		ctx,
+		s.NewRun(0).DefaultStartWorkflowOptions(),
+		"kitchenSink",
+		workflowParams,
+	)
+	if err != nil {
+		return fmt.Errorf("failed starting initial workflow: %w", err)
+	} else if err := workflowRun.Get(ctx, nil); err != nil {
+		return fmt.Errorf("failed executing initial workflow: %w", err)
+	}
+	resp, err := s.Client.DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+	if err != nil {
+		return fmt.Errorf("failed describing workflow: %w", err)
+	}
+	stateTransitionsPerWorkflow := resp.WorkflowExecutionInfo.StateTransitionCount
+	workflowStartInterval := durationPerStateTransition * time.Duration(stateTransitionsPerWorkflow)
+	s.Logger.Infof(
+		"Simple workflow takes %v state transitions, which means we need to start a workflow every %v. "+
+			"Running for %v now...",
+		stateTransitionsPerWorkflow,
+		workflowStartInterval,
+		s.Configuration.Duration,
+	)
+
+	// Start a workflow every X interval until duration reached or there are N
+	// start failures in a row
+
+	const maxConsecutiveErrors = 5
+	errCh := make(chan error, 10000)
+	ticker := time.NewTicker(workflowStartInterval)
```

Comment:
> the solution would need to also look at a metric and apply some control loop on top of that.

I intentionally didn't want to alter anything based on the metric because that's a bigger deal. I don't think the mandate is to maintain a steady count in the face of failures.